addons/website/static/src/js/show_password.js

```diff
@@ -0,0 +1,26 @@
+//
+// This file is meant to determine the timezone of a website visitor
+// If the visitor already exists, no need to keep the timezone cookie
+// as the timezone is set on the visitor.
+//
+odoo.define('website.show_password', function (require) {
+'use strict';
+
+var ajax = require('web.ajax');
+var utils = require('web.utils');
+var publicWidget = require('web.public.widget');
+
+publicWidget.registry.show_password = publicWidget.Widget.extend({
+    selector: '#showPass',
+
+    start: function () {
+        $( "#showPass" )
+            .mouseup(() => $("#password").attr('type', 'password'))
+            .mousedown(() => $("#password").attr('type', 'text'));
```

Comment:
You should use the `event` key of the public widget to define those handlers, but in fact, reading the code, I think you will have a bug if you mousedown -> move your cursor away -> mouseup, this will stay in text mode.